components/moderation-panel/ModEditSubmissionForm.vue

```diff
@@ -1199,66 +548,76 @@ async function submitUpdatedSubmission(e: Event) {
     // This updates the submission in the form with the values stored in the db on success
     if (submissionResult) initializeSubmissionFormValues(submissionResult)
     toast.success(t('modSubmissionForm.successMessageUpdated'))
-    if (moderationSubmissionStore.updatingSubmissionFromTopBar) {
-        router.push('/moderation')
+    if (moderationSubmissionStore.updatingSubmissionFromTopBarAndExiting) {
```

Comment:
nit: this PR is too big at this point, but we already knew where this was going 😆 🚀